### PR TITLE
Handle scheduling of account transfers

### LINKED ISSUE #76: Scheduled transfers don't get added on destination accounts
* Create a transfer from account A to account B with a date some months back
* Do a "Schedule this transaction" for a monthly repeat

The amounts get correctly removed from account A, but are not added to account B.

### PR DESCRIPTION
**Pull request by dospuntos**
Fixes #76 .

* Scheduling of transfers is now possible
* Trying to schedule the the transaction in the destination account will display an error message
* Scheduling a transfer generated by the scheduling script will display an error message
* Deleting one of the accounts will remove all related schedules
* Generated catkeys

To test:
- Have two accounts (e.g. "Source" and "Destination")
- Create a transfer from "Source" to "Destination", e.g. $100.
- Try to schedule the generated transfer in "Destination". An error message will be displayed
- Schedule the generated transfer in "Source" and set the date a few months back.
- The scheduled transfers will show up in both accounts.
- Try scheduling one of the newly created transfers. An error message will be displayed.

### CODE REVIEW

**Comment by humdingerb** [comment]
Nice! However, I did as you suggested with "To test", transferring from my first account to the second account, then scheduling that as "monthly" a few months back. it gets scheduled alright, but the I see both transactions in the first account, i.e. the amount gets substracted and added to the first account.

> Try to schedule the generated transfer in "Destination". An error message will be displayed

Would it be possible to simply open the transferring account for scheduling automatically, without showing the alert? For example, if you transfer from 1st to 2nd account, then try to schedule the transfer-transaction of the 2nd account, the app automatically opens the transfer-transaction of the 1st account? Because that's obviously what the user wanted.

File: src/MainWindow.cpp [hunk 1]
```diff
@@ -493,12 +493,23 @@ MainWindow::MessageReceived(BMessage* msg)
...
+						B_TRANSLATE(
+							"Transfers can only be scheduled from the originating account. This "
+							"appears to be the destination account for the transfer."));
+					break;
+				}
+
+				int32 destAccount = gDatabase.GetTransferDestination(data.GetID(), acc->GetID());
+				if (destAccount < 0) {
+					ShowAlert(B_TRANSLATE("Scheduling error"),
+						B_TRANSLATE("Transfers generated by scheduling cannot be scheduled."));
```
Comment (humdingerb) [inline src/MainWindow.cpp:510]: I think adding "themselves" would make it even clearer:

```suggestion
						B_TRANSLATE("Transfers generated by scheduling cannot be scheduled themselves."));
```

Comment (dospuntos) [reply src/MainWindow.cpp]: Yes, that's better. Added.

**Comment by humdingerb** [comment]
That fixed the issue, nice!

One last thing though. It may not occur with usual usage, where scheduling is mostly from now into the future, not back-dated... But when doing this:
* Enter a transfer from accountA to accountB.
* Set any date, say 01.07.2024 (1st of July).
* Now schedule that transfer to be monthly, and a start date of 01.03.2024. (first of March).
As a result, there are now two transfers. The original on 1st of July that was used for the scheduling and the monthly scheduled one.
This is a general issue with all transactions, not just transfers.

**Comment by dospuntos** [comment]
> As a result, there are now two transfers. The original on 1st of July that was used for the scheduling and the monthly scheduled one.

Wouldn't this be as expected? I think it would be strange if the original transaction would disappear without notice. Once scheduled, there's no relationship with the original transaction/transfer, so the user can delete the original transaction if needed without affecting the scheduling.

Maybe a solution would be to add a checkbox to the scheduling window? "Delete original transaction?" or similar?

**Comment by humdingerb** [comment]
Yeah, I expected this to be hard to handle...
It's not a big problem, I think, because the user will probably see the duplicate popping up right by the transaction they just used for scheduling. Also, normally people schedule stuff in the future, i.e. they create the original transaction on today, 15.07.2024, then schedule a monthly from next month on (15.08.2024).

I just mentioned it, because I noticed it.

> Maybe a solution would be to add a checkbox to the scheduling window? "Delete original transaction?" or similar?

Nah. I think that would confuse more than help...

Ready to merge?

**Comment by dospuntos** [comment]
Yes, you can merge this. Thanks

**Comment by humdingerb** [comment]
👍

**Pull request merged**